Src/WindowsServer/WindowsServer.Shared.Tests/AzureWebAppRoleEnvironmentTelemetryInitializerTest.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer
+{
+    using System;
+    using System.Globalization;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    using Microsoft.ApplicationInsights.WindowsServer.Azure;
+    using Microsoft.ApplicationInsights.WindowsServer.Azure.Emulation;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation;
+
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Assert = Xunit.Assert;
+    
+    [TestClass]
+    public class AzureWebAppRoleEnvironmentTelemetryInitializerTest
+    {
+        [TestMethod]
+        public void AzureWebAppRoleEnvironmentTelemetryInitializerSetsRoleName()
+        {
+            var telemetryItem = new EventTelemetry();
+
+            Environment.SetEnvironmentVariable("WEBSITE_SITE_NAME", "TestRoleName");
+            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", "TestRoleInstanceName");
+
+            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer();
+            initializer.Initialize(telemetryItem);
+
+            Assert.Equal("TestRoleName", telemetryItem.Context.Cloud.RoleName);
+            Assert.Equal("TestRoleInstanceName", telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Equal("TestRoleInstanceName", telemetryItem.Context.GetInternalContext().NodeName);
+
+            Environment.SetEnvironmentVariable("WEBSITE_SITE_NAME", null);
+            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", null);
```

Comment:
i'm following existing pattern in these tests ) in general i agree, but these vars are really specific for azure web apps, so it should not be the problem.